packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java

```diff
@@ -49,7 +52,18 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
       case "setShortcutItems":
         List<Map<String, String>> serializedShortcuts = call.arguments();
         List<ShortcutInfo> shortcuts = deserializeShortcuts(serializedShortcuts);
-        shortcutManager.setDynamicShortcuts(shortcuts);
+        ThreadPoolExecutor executor =
+            new ThreadPoolExecutor(0, 1, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+
+        executor.execute(
+            new Runnable() {
+              @Override
+              public void run() {
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N_MR1) {
+                  shortcutManager.setDynamicShortcuts(shortcuts);
```

Comment:
Right now you are calling result after you've dispatched the message to the background thread.  You should call the result after the background task has executed (calling `result` from the background thread should be thread-safe now, no need to dispatch back to the main thread).

Reply:
> calling `result` from the background thread should be thread-safe now

Does "now" include stable? Plugins support at least back to last stable (often more, but currently for this plugin it's 2.5)

Reply:
No, it doesn't I corrected myself over chat.  We can't call the result directly until background platform channels is on stable.